### DISCUSSION TITLE
USB query keyboard LEDS

### DISCUSSION THREAD

**Discussion by maotek** [stm32duino/Arduino_Core_STM32#2263] [Libraries] [answered]
Is there a way to get keyboards leds (capslock, numlock, scrolllock) using this stm32duino core? I am trying to get the caps lock state from the host. Im not very familiar with USB, but digging around in the stm32duino core, I found that DataOut is NULL? is there any other way to get the keyboard leds? maybe with clark's core? https://github.com/stm32duino/Arduino_Core_STM32/blob/main/cores/arduino/stm32/usb/hid/usbd_hid_composite.c#L118

**Comment by fpistm** [comment, marked as answer]
Currently not, the keyboard library is based on the official one which do not support this yet:
https://github.com/arduino-libraries/Keyboard/issues/43
